From dcc85e9173c92c43b618b262a61b95a28f8db036 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 19 Apr 2023 09:54:39 +0200
Subject: [PATCH 0485/1695] clk: stm32mp25: put BKPSRAM clock as critical

BKPSRAM is used by TF-M for its internal secure storage. This clock
has to be critical for now.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4055d47fdc3c53545b5695d0dadfb98210f02a94 clk: stm32mp25: put BKPSRAM clock as critical]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I01188f1b038d358fe033204a2cb3891f310947e8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/301488
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Ludovic BARRE <ludovic.barre@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/303914
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Change-Id: I8621dcb69d3d182d69320ad7291aa43a1488d0b5
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/405649
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 300dde8a0..ce57f984f 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -4018,6 +4018,7 @@ static bool clk_stm32_clock_is_critical(__maybe_unused struct clk *clk)
 		&ck_icn_p_ddrcfg,
 		&ck_icn_p_ddrphyc,
 		&ck_icn_s_sysram,
+		&ck_icn_s_bkpsram,
 		&ck_ker_ospi1,
 		&ck_ker_ospi2,
 		&ck_icn_s_vderam,
-- 
2.47.2

